### PR TITLE
Add com.stremio.Service

### PR DESCRIPTION
**Pull request by tymmesyde**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by tymmesyde** [comment]
bot, build com.stremio.Service

**Comment by tymmesyde** [comment]
bot, build com.stremio.Service

**Review by core1024 — APPROVED** [review]

File: com.stremio.Service.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{
+    "app-id": "com.stremio.Service",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "43",
```
Comment (hfiguiere) [inline com.stremio.Service.json:4]: 44 is the current runtime.

File: flathub.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+    "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: this is disappointing

File: com.stremio.Service.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            "install -Dm644 ./flatpak/com.stremio.Service.appdata.xml /app/share/metainfo/com.stremio.Service.appdata.xml"
+        ],
+        "sources": [
+            {
+                "type": "git",
+                "url": "https://github.com/Stremio/stremio-service.git",
+                "tag": "74143f710c3a4b668aee3a08da586e245405757f"
+            },
+            "./cargo-sources.json",
+            "./server-source.json"
```
Comment (hfiguiere) [inline com.stremio.Service.json:48]: there is strictly no reason to have a separe "server-source.json"

File: flathub.json (hunk 2 above)
Comment (tymmesyde) [reply flathub.json:2]: Can you elaborate ?

File: com.stremio.Service.json (hunk 3 above)
Comment (tymmesyde) [reply com.stremio.Service.json:48]: It's just easier for us to do that at the moment as we use a build script on our repo to generate this file, there is your reason

File: flathub.json (hunk 2 above)
Comment (hfiguiere) [reply flathub.json:2]: why do you disable aarch64?

Comment (tymmesyde) [reply flathub.json:2]: Because we don't support it at the moment, is this relevant to the app submission process ?

**Comment by tymmesyde** [comment]
bot, build com.stremio.Service

File: flathub.json (hunk 2 above)
Comment (hfiguiere) [reply flathub.json:2]: we offer flathub package on both architectures.

Comment (tymmesyde) [reply flathub.json:2]: We just don't support it at the moment

**Comment by barthalion** [comment]
bot, build com.stremio.Service

**Comment by barthalion** [comment]
```
~ [1] flatpak run com.stremio.Service
bwrap: execvp stremio-service: Permission denie
```

**Comment by barthalion** [comment]
As a someone who doesn't use Stremio, it's unclear to me how to use it even if I start it with the desktop file.

**Comment by hfiguiere** [comment]
> ```
> ~ [1] flatpak run com.stremio.Service
> bwrap: execvp stremio-service: Permission denie
> ```

this because
`"install -Dm755 server.js -t /app/bin/stremio-service/",`

ie `stremio-service` is a directory

File: com.stremio.Service.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                "CARGO_HOME": "/run/build/service/cargo"
+            }
+        },
+        "build-commands": [
+            "cargo --offline fetch --manifest-path Cargo.toml --verbose",
+            "cargo --offline build --release --features=bundled,offline-build --verbose"
+        ],
+        "post-install": [
+            "install -Dm755 server.js -t /app/bin/stremio-service/",
+            "install -Dm755 ./target/release/stremio-service -t /app/bin/stremio-service/",
```
Comment (hfiguiere) [inline com.stremio.Service.json:33]: this make `stremio-service` no found what is found from `$PATH` directory is the directory with that name.

I always frown when I see directories copied into the `bin` directory

Comment (tymmesyde) [reply com.stremio.Service.json:33]: That's why we were using a path in the command property 
The previous build worked

Comment (hfiguiere) [reply com.stremio.Service.json:33]: the command should through `PATH`. It is not.

Comment (tymmesyde) [reply com.stremio.Service.json:33]: What does this mean ?
Can you provide a clear explanation as to what to do to resolve this ?

Comment (hfiguiere) [reply com.stremio.Service.json:33]: the binary should be in `/app/bin`. Or a wrapper script to call it. And that wrapper should be the name of the executable is both `command` and the `.desktop` file.

**Comment by tymmesyde** [comment]
bot, build com.stremio.Service

File: com.stremio.Service.json (hunk 4 above)
Comment (tymmesyde) [reply com.stremio.Service.json]: It has been updated to use a wrapper

**Pull request closed without merging**